### PR TITLE
Adding user logging, observable creation and search logging

### PR DESCRIPTION
**Pull request by alexandrelim**
This enhancements provides :
- user logging 
- logging attempt
- user's password change
- observable creation logging 
- observable search logging

### CODE REVIEW

**Comment by tomchop** [comment]
Hey, thanks a lot for this PR! It's exactly what I had in mind when I mentioned user logging. It looks great.

While I fix the travis build (which seems to be failing because of some dependencies), there's a small change I'd like you to make on this.

Currently the file `logger.py` logs to is hardcoded, do you think you could fetch that information from the config file instead? Check out [this file](https://github.com/yeti-platform/yeti/blob/master/contrib/feeds/otx/otx.py#L9) for an example on how to use it.

**Comment by tomchop** [comment]
Just fixed the Travis problem in  #311, so feel free to merge master into your branch!

**Comment by alexandrelim** [comment]
Oh glad to hear that.
I'll work on this asap!

**Comment by alexandrelim** [comment]
Done.
Do you have any other request on this subject?

File: core/config/config.py [hunk 1]
```diff
@@ -60,3 +60,4 @@ def get(self, section, key, default=None):
 yeti_config.set_default_value('redis', 'database', 0)
 yeti_config.set_default_value('proxy', 'http', None)
 yeti_config.set_default_value('proxy', 'https', None)
+yeti_config.set_default_value('logging', 'filename', 'userlogs')
```
Comment (tomchop) [inline core/config/config.py]:
```suggestion
yeti_config.set_default_value('logging', 'filename', '/var/log/yeti.log')
```

**Comment by tomchop** [comment]
This looks good, thanks! I added a suggestion to make the default log path to `/var/log/yeti.log`.

For future consideration (maybe in another PR? 😃)

Also, I think it might be a good idea to consider making this an optional functionality. (logs can get big fast depending on system usage, and some people might not want queries to show up in their system log files). 

I think an easy way to do this would be, instead of calling `userLogger.something` to have a custom function in `logger.py` that takes in the log level + log message and decides to do the actual logging depending on a variable (this variable could very well be `yeti_config.get('logging', 'filename')` if it is set to `None` for instance.

**Comment by tomchop** [comment]
Hey @alexandrelim, with respect to my last comment, is this something you're still planning to work on?

**Comment by alexandrelim** [comment]
Hi,
Yes, sorry, it has been rough lately but I'll be working on it soon. :)

Le ven. 25 janv. 2019 à 14:01, Thomas Chopitea <notifications@github.com> a
écrit :

**Comment by tomchop** [comment]
No problem at all, just checking-in :)

**Comment by alexandrelim** [comment]
Hey @tomchop, 
That's is, I think. Sorry again for the delay. 
The only way I found to make this work was to make the user create a folder in /var/log and change its ownership. 
If you don't have any other request, I'll now work on the second version and make this feature optional.

**Comment by tomchop** [comment]
Awesome, I think we're almost there. The travis build [is failing](https://travis-ci.org/yeti-platform/yeti/builds/491278343#L775) because `/var/log/yeti/` can't be found. Can you please add the `mkdir` and `chown` statements from your documentation to:

 * the [TravisCI file](https://github.com/yeti-platform/yeti/blob/master/.travis.yml)
 * the [bootstrap](https://github.com/yeti-platform/yeti/tree/master/extras) scripts
 * the [Docker](https://github.com/yeti-platform/yeti/tree/master/extras/docker) scripts

After that, I think everything will be good to merge :)

**Comment by tomchop** [comment]
Alright, I updated things myself. Thanks a lot for your contribution!

**Pull request merged**